main.ts

```diff
@@ -51,15 +44,19 @@ export default class MyPlugin extends Plugin {
 
 	async loadSettings() {
 		this.settings = Object.assign(
-			{},
-			DEFAULT_SETTINGS,
-			await this.loadData(),
+			Object.assign(DEFAULT_SETTINGS, (await this.loadData()) ?? {})
 		);
 	}
 
 	async saveSettings() {
 		await this.saveData(this.settings);
 	}
+
+	/** Update plugin settings. */
+	async updateSettings(settings: Partial<PluginSettings>) {
+		Object.assign(this.settings, settings);
+		await this.saveData(this.settings);
+	}
```

Comment:
Reference for this addition: https://github.com/blacksmithgu/obsidian-dataview/blob/master/src/main.ts#L290